icefoganalytics/elcc-data-management
# ELCC-5: Establish Appropriate TimeStamp Columns In Database

Fixes ELCC-5

# Context

In the interest of basic data tracking, I'm standardizing the database by adding a created_at, update_at column to each table in the database. Its a common pattern that makes debugging data quality issues and analytics much easier. In the future we may also want to go the "paranoid" route and add a "deleted_at" column as well.

# Implementation

Tweak the umzug migrator so migrations run in a transaction.
Make `underscores` and `timestamps` the default in all sequelize models.
Add timestamp columns to umzug `SequelizeMeta` table, but don't enable timestamps until migration has run in all environments.

# Testing Instructions

1. Boot the app via `dev up`

2. Check that the app complies, and that you can log in at http://localhost:8080.

3. Check that the migrations ran and that all tables now have `created_at` and `deleted_at` columns.
